test/builtin_dataset_mocks.py

```diff
@@ -390,15 +390,17 @@ def create_ann_folder(root, name, file_name_fn, num_examples):
     images_root = root / "101_ObjectCategories"
     anns_root = root / "Annotations"
 
-    ann_category_map = {
-        "Faces_2": "Faces",
-        "Faces_3": "Faces_easy",
-        "Motorbikes_16": "Motorbikes",
-        "Airplanes_Side_2": "airplanes",
+    image_category_map = {
```

Comment:
The map was inversed before and did nothing.

Reply:
So the annotation folder had the same name as the image folder.  I'm wondering why the test didn't fail then. Is it because the dataset code doesn't actually care about the annotation folder names?

Reply:
To give some context: Caltech101 uses folders to separate the categories. The folder names match between the images and annotations except for four exceptions. The implementation matches the annotations folder names with this map

https://github.com/pytorch/vision/blob/08cc9a7fb5baa9655fa5d2dd44a6fcfcbe48a1c7/torchvision/prototype/datasets/_builtin/caltech.py#L45-L50

and this logic

https://github.com/pytorch/vision/blob/08cc9a7fb5baa9655fa5d2dd44a6fcfcbe48a1c7/torchvision/prototype/datasets/_builtin/caltech.py#L72-L73

The old mock data used the already matched names for the annotations so the dataset didn't need to do any matching. This means that the test were passing fine, although the implementation might be broken (it wasn't), since the matching was never tested.